src/proto/proto_common.cpp

```diff
@@ -511,6 +511,11 @@ void init_data_readers(
         reader_validation->get_data_store_ptr()->compact_nodes();
       }
 
+      size_t ntrain = reader->get_num_data();
+      if (ntrain == 0) {
+        LBANN_ERROR("num train samples = 0; something is wrong");
```

Comment:
@bvanessen I just want to double-check that we have no use cases where there are no training samples.

Reply:
My take from today's weekly mtg: OK to throw exception for now; revisit if someone runs an experiment where there's a train reader and the reader should have zero samples.